### DISCUSSION TITLE
salt fails after installation reboot

### DISCUSSION THREAD

**Discussion by trocade** [Security-Onion-Solutions/securityonion#3867] [Unsupported Versions] [answered]
So I just installed a new sensor, everything went just fine during install but after reboot it is stuck in WAIT
and when running salt on it I get the following error:
```
salt <sensor> state.apply
<sensor>_sensor:
    Data failed to compile:
----------
    Rendering SLS 'base:firewall' failed: Jinja variable list object has no element 0
/var/cache/salt/minion/files/base/firewall/hostgroups.yaml(23):
---
[...]
          - 127.0.0.1
    self:
      ips:
        delete:
        insert:
          - {{ salt['grains.get']('ip_interfaces').get(salt['pillar.get']('sensor:mainint', salt['pillar.get']('manager:mainint', salt['pillar.get']('elasticsearch:mainint', salt['pillar.get']('host:mainint')))))[0] }}    <======================
---
----------
    Rendering SLS 'base:filebeat' failed: Jinja variable list object has no element 0
ERROR: Minions returned with non-zero exit code
```

If I compare the pillar file to a working one I can not see any differences.

I also tried to do the get on all the values it tries to fetch, but I assume that only one of them need to return so it also seems to work?
```
salt <sensor>_sensor pillar.get sensor:mainint
<sensor>_sensor:
    eno1
salt <sensor> pillar.get manager:mainint
<sensor>_sensor:
salt <sensor>_sensor pillar.get elasticsearch:mainint
<sensor>_sensor:
salt <sensor>_sensor pillar.get host:mainint
<sensor>_sensor:
    eno1
```

A restart of the salt on the minion resolves the problem.

**Comment by TOoSmOotH** [comment, marked as answer]
This is a known issue that will be fixed in 2.3.50. It will delay the startup of the minion service. This is some weird race condition where the minion starts before networking is fully initialized and results in this condition. Restarting the minion fixes it. 

https://github.com/Security-Onion-Solutions/securityonion/issues/3543
